src/stable_graph.rs

```diff
@@ -279,6 +280,25 @@ impl<N, E, Ty, Ix> StableGraph<N, E, Ty, Ix>
         node_weight
     }
 
+    /// Retain only nodes from the graph if the `f` returns true and return the weights of the
+    /// removed nodes. If no nodes are removed, the Vec is empty.
+    ///
+    /// The node indecies of the removed nodes are invalidated, but none other.
+    /// Edge indices are invalidated as they would be following the removal of
+    /// each edge with an endpoint in a removed node.
+    ///
+    /// Computes in **O(n + e')** time, where **n** is the number of node indices and
+    ///  **e'** is the number of affected edges, including *n* calls to `.remove_edge()`
+    /// where *n* is the number of edges with an endpoint in a removed node.
+    pub fn retain_nodes<F>(&mut self, mut f: F) where F: FnMut(Frozen<Self>, NodeIndex<Ix>) -> bool {
+        for i in 0..self.g.node_count() {
+            let ix = node_index(i);
+            if !f(Frozen(self), ix) {
```

Comment:
The node count is correct: the count of the unstable subgraph is used. 
Added the check. The users should indeed not do this manually.